src/main/java/org/springframework/shell2/JLineShell.java

```diff
@@ -155,10 +165,7 @@ public void run() throws IOException {
 					result = ReflectionUtils.invokeMethod(method, methodTarget.getBean(), args);
 				}
 				catch (ExitRequest e) {
-					if (applicationContext instanceof Closeable) {
```

Comment:
Correct, using a ResultHandler instead of an exception. Not a strong opinion though, as the app will close anyways :)